pkg/webhooks/utils/block.go

```diff
@@ -43,8 +43,16 @@ func GetBlockedMessages(engineResponses []engineapi.EngineResponse) string {
 	for _, er := range engineResponses {
 		ruleToReason := make(map[string]string)
 		for _, rule := range er.PolicyResponse.Rules {
-			if rule.Status() != engineapi.RuleStatusPass {
-				ruleToReason[rule.Name()] = rule.Message()
+			if len(engineResponses) > 1 {
+				if rule.Status() != engineapi.RuleStatusPass {
+					if rule.Message() != "preconditions not met" {
```

Comment:
It is better to check that the rule status isn't equal to `skip` because rules that have this message always get a `skip` result.

```
for _, rule := range er.PolicyResponse.Rules {
  if rule.Status() != engineapi.RuleStatusPass && rule.Status() != engineapi.RuleStatusSkip {
     ruleToReason[rule.Name()] = rule.Message()
  }
}
```

Reply:
Okay will update